lib/pbench/test/unit/agent/test_utils.py

```diff
@@ -0,0 +1,210 @@
+        assert bs.bind_port == bs.port
+        assert repr(bs) == "forty-two - localhost:2345"
+
+        bs = OurServer("127.0.0.42:4567", _def)
+        assert bs.pid_file is None
+        assert bs.host == "127.0.0.42"
+        assert bs.port == 4567
+        assert bs.bind_host == bs.host
+        assert bs.bind_port == bs.port
+        assert repr(bs) == "forty-two - 127.0.0.42:4567"
+
+        bs = OurServer("[127::42]:4567", _def)
+        assert bs.pid_file is None
+        assert bs.host == "127::42"
+        assert bs.port == 4567
+        assert bs.bind_host == bs.host
+        assert bs.bind_port == bs.port
+        assert repr(bs) == "forty-two - 127::42:4567"
+
+        bs = OurServer("bind.example.com:2345;host.example.com:6789", _def)
+        assert bs.pid_file is None
+        assert bs.host == "host.example.com"
+        assert bs.port == 6789
+        assert bs.bind_host == "bind.example.com"
+        assert bs.bind_port == 2345
+        assert repr(bs) == "forty-two - bind.example.com:2345 / host.example.com:6789"
+
+    def test_kill(self, pytestconfig, monkeypatch):
+        bs = OurServer("localhost", "localhost")
+        with pytest.raises(AssertionError):
+            bs.kill(1)
+
+        bs = OurServer("localhost", "localhost")
+        TMP = pathlib.Path(pytestconfig.cache.get("TMP", None))
+        pidfile = TMP / "test.pid"
+        pidfile.write_text("12345")
+        bs.pid_file = pidfile
+        ret = bs.kill(42)
+        assert ret == 42
+
+        pidfile.write_text("badpid")
+        ret = bs.kill(42)
+        assert ret == (BaseReturnCode.KILL_BADPID * 100) + 42
+
+        bs.pid_file = TMP / "enoent.pid"
+        ret = bs.kill(42)
+        assert ret == 42
+
+        class MockPath:
+            def __init__(self, kind):
+                self.kind = kind
+
+            def read_text(self):
+                if self.kind == "os":
+                    raise OSError("fake oserror")
+                else:
+                    raise Exception("fake exception")
+
+        bs.pid_file = MockPath("os")
+        ret = bs.kill(42)
+        assert ret == 342
+
+        bs.pid_file = MockPath("exc")
+        ret = bs.kill(42)
+        assert ret == 142
+
+        class MockTime:
+            def __init__(self):
+                self._clock = 0
+
+            def time(self, *args, **kwargs):
+                self._clock += 1
+                return self._clock
+
+            def sleep(self, *args, **kwargs):
+                return
+
+        class MockKill:
+            behaviors = {
+                "1001": (True, True, False, False, False, False),
+                "1002": (False, False, False, False, False, False),
+                "1003": (True, True, False, True, False, False),
+                "1004": (True, False, False, True, False, False),
+                "1005": (True, False, False, False, True, False),
+                "1006": (True, False, False, False, False, True),
```

Comment:
The third column ("killed by term"?) is always `False`.  Are you missing a test case, or should this column be omitted from the table?

Reply:
The first case was incorrect, fixed in a separate commit.